File: client-udp.c
/*
Disciplina: Teleprocessamento e Redes
Ano/Período: 2010/1
Prof.: Silvana Rossetto
Programa básico de um cliente UDP
*/

#include <stdlib.h>
#include <stdio.h>
#include <errno.h>
#include <string.h>
#include <sys/socket.h>
#include <sys/types.h>
#include <netinet/in.h>
#include <unistd.h> 

#define BUFFERSIZE 100
 
int main(int argc, char *argv[]) {
  int sock;
  struct sockaddr_in addr;
  int bytes_sent, buffer_length;
  char buffer[BUFFERSIZE];
  char *servIP;                
  unsigned short servPort;   
 
  // verifica se o IP e numero da porta do processo servidor foi informado
  if (argc != 3) {
      printf("--Erro na inicializacao: client <remote server IP> <remote server Port>\n");
      exit(0);
  }
  servIP = argv[1];
  servPort = atoi(argv[2]);

  // cria o descritor de socket para o servico entrega nao-confiavel
  if ((sock = socket(PF_INET, SOCK_DGRAM, IPPROTO_UDP)) < 0) {
    printf("--Erro na criacao do socket\n");
    exit(0);
  }
 
  // configura a estrutura de dados com o endereco local 
  memset(&addr, 0, sizeof(addr));
  addr.sin_family = AF_INET;
  addr.sin_addr.s_addr = inet_addr(servIP); 
  addr.sin_port = htons(servPort);
 
  printf("digite uma mensagem: \n"); 
  scanf(" %[^\n]",buffer);
  bytes_sent = sendto(sock, buffer, strlen(buffer), 0, (struct sockaddr*)&addr, sizeof (struct sockaddr_in));
  if (bytes_sent < 0) {
     printf("--Erro no recebimento \n");
  }
  close(sock); 
  return 0;
}

File: aula3/client-udp.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# vi:ts=4 sw=4

# This is a Python version of cods-aula-3/client-udp.c

import sys
import socket

#from __future__ import with_statement # This isn't required in Python 2.6
#from contextlib import closing


def main(argv):
    if len(argv) != 3:
        sys.stderr.write("Erro na inicializacao: client <remote server IP> <remote server Port>\n")
        sys.exit(1)

    servIP = argv[1]
    servPort = int(argv[2])

    #sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_UDP)
    # The last parameter can be omitted:
    sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)

    addr = (servIP, servPort)

    buffer = raw_input("digite uma mensagem:\n")
    bytes_sent = sock.sendto(buffer, addr)
    # I could have specified the flags:
    # socket.sendto(buffer, 0, addr)

    sys.stdout.write("Bytes sent: %d\n" % (bytes_sent,))
    sock.close()


if __name__ == '__main__':
    main(sys.argv)
